src/aead/aes.rs

```diff
@@ -449,7 +511,22 @@ fn detect_implementation(cpu_features: cpu::Features) -> Implementation {
     )))]
     let _cpu_features = cpu_features;
 
-    #[cfg(target_arch = "aarch64")]
+    #[cfg(all(target_arch = "aarch64", target_os = "linux"))]
+    {
+        match aes_support() {
+            Ok(aes_support) => {
+                if cpu::arm::AES.available(cpu_features) && aes_support {
```

Comment:
We cannot open a file and read from it every time we try to encrypt/decrypt something.

Instead, change the implementation of `cpu::arm` so that `cpu::arm::AES.available(cpu_features)` returns false. This would either require changing the implementation of `detect_features()` if `getauxval` is the problem, or the way `ARMCAP_STATIC` is initialized, if `cfg!(target_feature = "aes")` is the problem.

But, first, we need a clearer understanding of what is the actual cause of the problem. I feel like this PR is just wallpapering over the issue.